plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/testing/ResourceHudiTablesInitializer.java

```diff
@@ -91,18 +100,44 @@
 public class ResourceHudiTablesInitializer
         implements HudiTablesInitializer
 {
+    private static final Logger log = Logger.get(ResourceHudiTablesInitializer.class);
+    private static final String HASH_ALGORITHM = "SHA-256";
     private static final String TEST_RESOURCE_NAME = "hudi-testing-data";
 
+    // There might be other entry points that are using this initializer, make the location unique so it is more identifiable via logs
+    private final String baseLocationPrefix = UUID.randomUUID().toString();
+    private final Path tempDir;
+
+    /**
+     * Manually declaring a temp directory here and performing a manual cleanup as this constructor is invoked in HudiQueryRunner in a @BeforeAll static function.
+     * This means that jupiter's managed @TempDir annotation cannot be used as the path will be passed as null.
+     */
+    public ResourceHudiTablesInitializer()
+    {
+        // There are multiple entry points and they may perform unzipping together, ensure that they are all unzipping to different paths
+        try {
+            this.tempDir = Files.createTempDirectory(TEST_RESOURCE_NAME + "_" + baseLocationPrefix);
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     @Override
     public void initializeTables(QueryRunner queryRunner, Location externalLocation, String schemaName)
             throws Exception
     {
-        HudiTableUnzipper.unzipAllItemsInResource(TEST_RESOURCE_NAME);
+        // Inflate all deflated test resource archives to a temporary directory
+        HudiTableUnzipper.unzipAllItemsInResource(TEST_RESOURCE_NAME, tempDir);
```

Comment:
If there is concurrent unzipping happening, it might be better to have flag inside this initializer to determine if the test tables are populated / unzipped; if not, go through a synchronized block that does the unzipping to avoid corrupted tables being generated.  We can take this as a follow-up.